### PR TITLE
Updating data and documentation files

### PR DESCRIPTION
**Pull request by m-sezgin**

### CODE REVIEW

**Comment by beanumber** [comment]
As per our conversation Monday are you planning to make another push to this PR @m-sezgin ?

**Comment by m-sezgin** [comment]
just pushed!

File: R/data.R [hunk 1]
```diff
@@ -1,5 +1,52 @@
...
 #' @docType data
+#' @format
+#' An object of class \code{tbl_df} (inherits from \code{tbl}, \code{data.frame}) with 328141 rows and 45 columns.
+#' \describe{
+#'   \item{issuing_agency}{Boston Police division that issuing officer belongs to. a character vector.}
+#'   \item{agency_code}{Shorthand code for issuing agency. a character vector.}
+#'   \item{officer_id}{Unique identifier for BPD offficers. a character vector.}
+#'   \item{officer_name}{The name of the issuing officer.}
+#'   \item{event_date}{Year-Month-Day.}
+#'   \item{time_hh}{Hour of incident.}
```
Comment (beanumber) [inline R/data.R]: What do you think about converting these four variables into a timestamp? Wouldn't that be more efficient storage-wise?

File: R/data.R [hunk 2]
```diff
@@ -1,5 +1,52 @@
...
+#'   \item{time_mm}{Minute of incident.}
+#'   \item{am_pm}{a character vector}
+#'   \item{viol_type}{The type of violator. Owner, operator, passenger, or bike.}
+#'   \item{citation_number}{"unique identifier for the citation; the same number will appear multiple times in the spreadsheet if the ticket is for multiple offenses." (Woke Windows)}
+#'   \item{citation_type}{Can be civil, criminal, or arrest.}
+#'   \item{offense}{"code for an offense; seems to correspond to a section in the Massachusetts General Laws e.g. '90160' is MGL c.90 §16" (Woke Windows).}
```
Comment (beanumber) [inline R/data.R]: Can we add a link to this? 

https://malegislature.gov/Laws/GeneralLaws/PartI/TitleXIV/Chapter90/Section16

File: R/data.R [hunk 3]
```diff
@@ -1,5 +1,52 @@
...
+#'   \item{offense_description}{Detailed description of offense.}
+#'   \item{disposition}{Code for result of offense.}
+#'   \item{disposition_desc}{a character vector}
```
Comment (beanumber) [inline R/data.R]: But what does it mean?

File: R/data.R [hunk 4]
```diff
@@ -1,5 +1,52 @@
...
+#'   \item{location_name}{a character vector}
```
Comment (beanumber) [inline R/data.R]: The location in which the offense occurred

File: R/data.R [hunk 5]
```diff
@@ -1,5 +1,52 @@
...
+#'   \item{searched}{Whether a search was conducted.}
+#'   \item{crash}{Whether the incident was a collision.}
+#'   \item{court_code}{"see \href{https://www.mass.gov/info-details/trial-court-codes-numerical-listing}{Trial Court codes} for mapping" (Woke Windows)}
```
Comment (beanumber) [inline R/data.R:25]: Hmm...should we add these as helper data frames?

File: R/data.R [hunk 6]
```diff
@@ -1,5 +1,52 @@
...
+#'   \item{gender}{a character vector}
+#'   \item{year_of_birth}{Birthdate of violator.}
+#'   \item{lic_state}{State in which driver is licensed.}
+#'   \item{lic_class}{See \href{https://www.mass.gov/files/documents/2018/03/22/chapter_1_0.pdf}{Massachussets Driver's Manual} for more information on license classes.}
+#'     \item{cdl}{Whether the driver's license is a Commercial Driver's License.}
+#'     \item{plate_type}{This is likely plate type as specified \href{https://www.mass.gov/doc/passenger-plates-manual/download}{here}}
```
Comment (beanumber) [inline R/data.R:31]: Likely?

File: R/data.R [hunk 7]
```diff
@@ -1,5 +1,52 @@
...
+#'     \item{haz_mat}{Whether hazardous materials were present upon search.}
+#'     \item{amount}{Dollar fee associated with citation.}
+#'     \item{paid}{Whether the citation was paid.}
+#'     \item{hearing_requested}{Did the defendant request a hearing.}
+#'     \item{speed}{Whether the citation was speed-related.}
+#'     \item{posted_speed}{Posted speed limit in miles per hour (MPH).}
+#'     \item{viol_speed}{Speed in MPH. Provided if citation is speed-related.}
+#'     \item{posted}{Possibly whether a violater knew the posted speed.}
+#'     \item{radar}{Type of radar used to check speed.}
+#'     \item{clocked}{A character vector.}
```
Comment (beanumber) [inline R/data.R]: Indicating what?

File: R/data.R [hunk 8]
```diff
@@ -1,5 +1,52 @@
...
+#'     \item{officer_cert}{Likely how the citation was served on the violator.}
+#'     \item{race}{The race of the violator. Races with an insufficient number of entries were not included in this dataset. a character vector.}
```
Comment (beanumber) [inline R/data.R]: Isn't this omitted in favor of `race_clean`?

File: R/data.R [hunk 9]
```diff
@@ -1,5 +1,52 @@
 #' WokeWindows citations
 #' @docType data
+#' @format
+#' An object of class \code{tbl_df} (inherits from \code{tbl}, \code{data.frame}) with 328141 rows and 45 columns.
```
Comment (beanumber) [inline R/data.R]: Are these numbers still accurate?

File: data-raw/woke_windows.R [hunk 10]
```diff
@@ -30,12 +30,16 @@ boston_pd_1120 <- boston_pd_1120 |>
       race == "asian" ~ "ASIAN",
       race == "UNK" ~ "UNKNWN",
       TRUE ~ race)
-  )
+  ) |>
+  rename(disposition_desc = disposition_desc_15) |>
+  select(-c(disposition_desc_40, race)) |>
+  sample_n(150000)
```
Comment (beanumber) [inline data-raw/woke_windows.R]: I'm thinking now about whether a random sample is preferable to a set of years...

Comment (beanumber) [reply data-raw/woke_windows.R]: As per the comment above, we could add something like:

`mutate(event_timestamp = lubridate::parse_date_time(...))`

**Review by beanumber — CHANGES_REQUESTED** [review]
Let's talk about this later today.

**Comment by beanumber** [comment]
@m-sezgin are you planning to push any commits to this before our meeting this afternoon?

**Comment by m-sezgin** [comment]
I pushed but I'm running into some issues with court code documentation and datetimes

**Comment by m-sezgin** [comment]
Requested changes have been made. Only issues are that `clocked` and `plate_type` documentation are still unknown.

**Pull request merged**